### ISSUE TITLE
Gatsby build fails on shared hosting where listing of folder content (e.g /home) is not allowed

### ISSUE THREAD

**Issue by andrejpavlovic** [gatsbyjs/gatsby#14962]
## Description

I've tried running `gatsby build` on my shared hosting, but the build failed. More specifically, it seems to fail due to the fact that one of the parent directories (in this case `/home`) does not permit listing of files. The `true-case-path` library, used by gatsby, throws an error because it is unable to `scandir` a component path's parent folder (in this case the component path is `/home/myusername/gatsby-site/src/pages/404.js`). I think the build should proceed in this case, and handle this error silently.

### Steps to reproduce

```
cd /home/myusername
npm install -g gatsby
gatsby new gatsby-site
cd gatsby-site
npm run build
```

### Expected result

Build should complete successfully. If parent directories do not allow files to be listed, the build should still proceed.

### Actual result

```
[myusername@server gatsby-site]$ npm run build

> gatsby-starter-default@0.1.0 build /home/myusername/gatsby-site
> gatsby build

success open and validate gatsby-configs - 0.011 s
success load plugins - 0.491 s
success onPreInit - 0.010 s
success delete html and css files from previous builds - 0.010 s
success initialize cache - 0.021 s
success copy gatsby files - 0.043 s
success onPreBootstrap - 0.012 s
success source and transform nodes - 0.073 s
success building schema - 0.312 s
success createPages - 0.002 s
/home/myusername/gatsby-site/src/pages/404.js
error UNHANDLED REJECTION


  Error: EACCES: permission denied, scandir '/home'

  - sync.js:288 GlobSync._readdir
    [gatsby-site]/[glob]/sync.js:288:41

  - sync.js:137 GlobSync._processReaddir
    [gatsby-site]/[glob]/sync.js:137:22

  - sync.js:132 GlobSync._process
    [gatsby-site]/[glob]/sync.js:132:10

  - sync.js:207 GlobSync._processReaddir
    [gatsby-site]/[glob]/sync.js:207:10

  - sync.js:132 GlobSync._process
    [gatsby-site]/[glob]/sync.js:132:10

  - sync.js:48 new GlobSync
    [gatsby-site]/[glob]/sync.js:48:10

  - sync.js:26 Function.globSync [as sync]
    [gatsby-site]/[glob]/sync.js:26:10

  - index.js:29 trueCasePathSync
    [gatsby-site]/[true-case-path]/index.js:29:15

  - public.js:198 actions.createPage
    [gatsby-site]/[gatsby]/dist/redux/actions/public.js:198:41

  - redux.js:468
    [gatsby-site]/[redux]/lib/redux.js:468:35

  - api-runner-node.js:68 doubleBoundActionCreators.(anonymous function).args
    [gatsby-site]/[gatsby]/dist/utils/api-runner-node.js:68:20

  - gatsby-node.js:141 _createPage
    [gatsby-site]/[gatsby-plugin-page-creator]/gatsby-node.js:141:3

  - gatsby-node.js:76
    [gatsby-site]/[gatsby-plugin-page-creator]/gatsby-node.js:76:22

  - Array.forEach

  - gatsby-node.js:75 Object._callee$
    [gatsby-site]/[gatsby-plugin-page-creator]/gatsby-node.js:75:19
```

### Environment

```
  System:
    OS: Linux 3.10 CentOS Linux 7 (Core)
    CPU: (48) x64 Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz
    Shell: 4.2.46 - /bin/bash
  Binaries:
    Node: 10.15.1 - ~/webapps/nodejs/bin/node
    npm: 6.4.1 - ~/webapps/nodejs/bin/npm
  Languages:
    Python: 2.7.5 - /usr/local/bin/python
  npmPackages:
    gatsby: ^2.9.11 => 2.9.11
    gatsby-image: ^2.1.4 => 2.1.4
    gatsby-plugin-manifest: ^2.1.1 => 2.1.1
    gatsby-plugin-offline: ^2.1.3 => 2.1.3
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-sharp: ^2.1.9 => 2.1.9
    gatsby-source-filesystem: ^2.0.43 => 2.0.43
    gatsby-transformer-sharp: ^2.1.21 => 2.1.21
  npmGlobalPackages:
    gatsby: 2.10.0
```

**Comment by andrejpavlovic** [comment]
The workaround for now for me was to directly edit `gatsby-site/node_modules/gatsby/dist/redux/actions/public.js` and replace the following code:

https://github.com/gatsbyjs/gatsby/blob/08d5c05d4e7127fcb7fb3d608199d70b80209d0e/packages/gatsby/src/redux/actions/public.js#L215

with

```javascript
        let trueComponentPath = page.component;
        try {
          trueComponentPath = slash(truePath(page.component));
        } catch (e) {}
```

**Comment by waltercruz** [comment]
I'm facing this problem too!

**Comment by waltercruz** [comment]
maybe this issue should be handled on true-case-path? There's a issue opened, from a long time ago:

https://github.com/barsh/true-case-path/issues/2

**Comment by andrejpavlovic** [comment]
not stale

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.13.66`
